src/components/Articles/CommentAdder.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from "react";
+import { postCommentByArticleId } from "../../api";
+import {
+  TextField,
+  Button,
+  MenuItem,
+  Select,
+  CircularProgress,
+  Alert
+} from "@mui/material";
+
+export const CommentAdder = ({ article_id, setComments }) => {
+  const [newComment, setNewComment] = useState("");
+  // change below after auth set up
+  const [username, setUsername] = useState("");
+  const [isAddingComment, setIsAddingComment] = useState(false);
+  const [error, setError] = useState(null);
+
+  const users = [
+    "tickle122",
+    "grumpy19",
+    "happyamy2016",
+    "cooljmessy",
+    "weegembump",
+    "jessjelly",
+    "choose for error"
+  ];
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setIsAddingComment(true) //
+    postCommentByArticleId(username, newComment, article_id)
```

Comment:
Great work on this, Only thing I noticed is that I can currently post an empty comment so it would be great to add in some logic to stop this behaviour.